Tests missing check for overlay - fail on RHEL6

### Version of Singularity:

```
3.5.2
```

### Actual behavior

```
    --- FAIL: TestImport (0.00s)
        mount_linux_test.go:578: overlay filesystem not supported
        mount_linux_test.go:581: returned a wrong number of mount kernel mount points 0 instead of 4
        mount_linux_test.go:584: returned a wrong number of mount kernel mount points 0 instead of 3
    FAIL
    coverage: 82.4% of statements
    FAIL    github.com/sylabs/singularity/internal/pkg/util/fs/mount        0.016s
```

```
   === RUN   TestFromSif
    --- FAIL: TestFromSif (2.30s)
        bundle_linux_test.go:91: failed to create overlay: failed to mount /tmp/bundle203730219/overlay: no such device
    FAIL
    coverage: 57.1% of statements
```

--- a/internal/pkg/util/fs/mount/mount_linux_test.go
+++ b/internal/pkg/util/fs/mount/mount_linux_test.go
@@ -12,6 +12,7 @@ import (
 
 	specs "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/sylabs/singularity/internal/pkg/test"
+	"github.com/sylabs/singularity/internal/pkg/test/tool/require"
 )
 
 func TestImage(t *testing.T) {
@@ -540,6 +541,22 @@ func TestImport(t *testing.T) {
 	}
 	points.RemoveAll()
 
+	invalidSpecs := []specs.Mount{
+		{
+			Source:      "/image.simg",
+			Destination: "/tmp/image",
+			Type:        "squashfs",
+			Options:     []string{"nosuid", "nodev"},
+		},
+	}
+	if err := points.ImportFromSpec(invalidSpecs); err == nil {
+		t.Errorf("should have failed with non authorized filesystem type")
+	}
+	points.RemoveAll()
+
+	// The specs below require overlay, must skip on e.g. RHEL6
+	require.Filesystem(t, "overlay")
+
 	points = &Points{}
 
 	validSpecs := []specs.Mount{
@@ -574,6 +591,7 @@ func TestImport(t *testing.T) {
 			Options:     []string{"nosuid"},
 		},
 	}
+
 	if err := points.ImportFromSpec(validSpecs); err != nil {
 		t.Error(err)
 	}
@@ -583,19 +601,7 @@ func TestImport(t *testing.T) {
 	if len(points.GetByTag(UserbindsTag)) != 3 {
 		t.Errorf("returned a wrong number of mount kernel mount points %d instead of 3", len(points.GetByTag(UserbindsTag)))
 	}
-	points.RemoveAll()
 
-	invalidSpecs := []specs.Mount{
-		{
-			Source:      "/image.simg",
-			Destination: "/tmp/image",
-			Type:        "squashfs",
-			Options:     []string{"nosuid", "nodev"},
-		},
-	}
-	if err := points.ImportFromSpec(invalidSpecs); err == nil {
-		t.Errorf("should have failed with non authorized filesystem type")
-	}
 }
 
 func TestTag(t *testing.T) {

--- a/pkg/ocibundle/sif/bundle_linux_test.go
+++ b/pkg/ocibundle/sif/bundle_linux_test.go
@@ -20,6 +20,7 @@ import (
 	"github.com/sylabs/singularity/internal/pkg/client/cache"
 	"github.com/sylabs/singularity/internal/pkg/test"
 	testCache "github.com/sylabs/singularity/internal/pkg/test/tool/cache"
+	"github.com/sylabs/singularity/internal/pkg/test/tool/require"
 )
 
 func TestFromSif(t *testing.T) {
@@ -72,33 +73,51 @@ func TestFromSif(t *testing.T) {
 		t.Errorf("unexpected success while creating OCI bundle")
 	}
 
-	// create OCI bundle from SIF
-	bundle, err = FromSif(sifFile, bundlePath, true)
-	if err != nil {
-		t.Fatal(err)
+	tests := []struct {
+		name     string
+		writable bool
+	}{
+		{"FromSif", false},
+		{"FromSifWritable", true},
 	}
-	// generate a default configuration
-	g, err := generate.New(runtime.GOOS)
-	if err != nil {
-		t.Fatal(err)
-	}
-	// remove seccomp filter for CI
-	g.Config.Linux.Seccomp = nil
-	g.SetProcessArgs([]string{tools.RunScript, "id"})
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
 
-	if err := bundle.Create(g.Config); err != nil {
-		// check if cleanup occurred
-		t.Fatal(err)
+			if tt.writable {
+				require.Filesystem(t, "overlay")
+			}
+
+			// create OCI bundle from SIF
+			bundle, err = FromSif(sifFile, bundlePath, tt.writable)
+			if err != nil {
+				t.Fatal(err)
+			}
+			// generate a default configuration
+			g, err := generate.New(runtime.GOOS)
+			if err != nil {
+				t.Fatal(err)
+			}
+			// remove seccomp filter for CI
+			g.Config.Linux.Seccomp = nil
+			g.SetProcessArgs([]string{tools.RunScript, "id"})
+
+			if err := bundle.Create(g.Config); err != nil {
+				// check if cleanup occurred
+				t.Fatal(err)
+			}
+
+			// execute oci run command
+			args = []string{"oci", "run", "-b", bundlePath, filepath.Base(sifFile)}
+			cmd = exec.Command(sing, args...)
+			if err := cmd.Run(); err != nil {
+				t.Error(err)
+			}
+
+			if err := bundle.Delete(); err != nil {
+				t.Error(err)
+			}
+
+		})
 	}
 
-	// execute oci run command
-	args = []string{"oci", "run", "-b", bundlePath, filepath.Base(sifFile)}
-	cmd = exec.Command(sing, args...)
-	if err := cmd.Run(); err != nil {
-		t.Error(err)
-	}
-
-	if err := bundle.Delete(); err != nil {
-		t.Error(err)
-	}
 }

